### PR TITLE
Added `ApiRenderer::renderDefaultValue`, fixed #38

### LINKED ISSUE #38: docs: bad octal number representation
This issue has originally been reported by @hiqsol at https://github.com/yiisoft/yii2/issues/10369.
Moved here by @cebe.

---

That's what we see in [docs](http://www.yiiframework.com/doc-2.0/yii-helpers-basefilehelper.html#createDirectory%28%29-detail):

``` php
public static boolean createDirectory ( $path, $mode = 509, $recursive = true )
```

And in the source:

``` php
public static function createDirectory($path, $mode = 0775, $recursive = true)
```

### PR DESCRIPTION
**Pull request by hiqsol**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Fixed issues  | #38 

Checked the generated docs for `yiisoft/yii2`.
Found no other special constants there.

### CODE REVIEW

File: templates/html/ApiRenderer.php [hunk 1]
```diff
@@ -284,6 +284,28 @@ public function renderMethodSignature($method, $context = null)
...
+     * @return string
+     */
+    public function renderDefaultValue($value)
+    {
+        if ($value===null) {
+            return 'null';
+        }
+
+        static $specials = [
+            '509' => '0775',
```
Comment (cebe) [inline templates/html/ApiRenderer.php]: should also cover `0664`, `0644` and `0755`.

File: templates/html/ApiRenderer.php [hunk 2]
```diff
@@ -284,6 +284,28 @@ public function renderMethodSignature($method, $context = null)
...
+            '2113696' => '0x2040A0',
```
Comment (cebe) [inline templates/html/ApiRenderer.php]: what is this?

Comment (hiqsol) [inline templates/html/ApiRenderer.php]: https://github.com/yiisoft/yii2/blob/master/framework/captcha/CaptchaAction.php#L71

Comment (cebe) [inline templates/html/ApiRenderer.php]: I see, maybe a comment should point to where it is used :)

**Comment by cebe** [comment]
Merged, thank you!

**Pull request closed without merging**